internal/docker/builder.go

```diff
@@ -211,18 +221,18 @@ func (d *Builder) ConstructBlueprints(bs []b.Blueprint) error {
 }
 
 // construct all Homeservers sequentially then commits them
-func (d *Builder) construct(bprint b.Blueprint, bpWg *sync.WaitGroup) {
-	defer bpWg.Done()
-	networkID := createNetwork(d.Docker, bprint.Name)
-	if networkID == "" {
-		return
+func (d *Builder) construct(bprint b.Blueprint) (errs []error) {
```

Comment:
I'd just have `(err error)` as the response and fail-fast and bail out of this function the moment anything goes wrong. We don't need to aggregate errors from different homeserver images.

Reply:
I think it is better to try everything and then informing the user of all the failures (instead to have only one failure at a time, which can be very time consuming to debug).

Reply:
But if you prefer I can change this to fail early and ignore the rest.

Reply:
I don't have strong opinions either way. I generally lean on the side of "what makes the code simpler" than "what presents the most information to the user" when it comes to internal guts. For tests themselves then it's the other way around. I don't mind.